File: sirma-platform/ui2/src/services/rest/mailbox-info-service.js
import {Inject, Injectable} from 'app/app';
import {RestClient, HEADER_V2_JSON} from 'services/rest-client';
import _ from 'lodash';

const serviceUrl = '/mailbox';

@Injectable()
@Inject(RestClient)
export class MailboxInfoService {

  constructor(restClient) {
    this.restClient = restClient;
    this.config = {};
    this.config.headers = {
      'Accept': HEADER_V2_JSON,
      'Content-Type': HEADER_V2_JSON
    };
  }

  /**
   * Get unread messages count for all registered accounts
   * @param accountName - mailbox account
   * @param config - service configuration
   * @returns number of unread messages
   */
  getUnreadMessagesCount(accountName, config) {
    config = _.defaults(config || {}, this.config);
    return this.restClient.get(`${serviceUrl}/${accountName}/unread`, config);
  }

}

File: docker/sirma-platform/platform/seip-parent/collaboration/email-integration/email-integration-api/src/main/java/com/sirma/sep/email/service/MailboxInfoService.java
package com.sirma.sep.email.service;

import com.sirma.sep.email.exception.EmailIntegrationException;

/**
 * Service used to extract specific information about given mailbox
 * 
 * @author S.Djulgerova
 */
public interface MailboxInfoService {

	/**
	 * Get count of unread messages in given mailbox (including external accounts)
	 * 
	 * @param accountName
	 *            - mailbox account name
	 * @return count of unread messages
	 * @throws EmailIntegrationException
	 */
	int getUnreadMessagesCount(String accountName) throws EmailIntegrationException;

}